### PR TITLE
Add GATv2Conv

### PR DESCRIPTION
**Pull request by shakedbr**
Hi,

I added GATv2  from the ["How Attentive are Graph Attention Networks?"](https://arxiv.org/abs/2105.14491) paper, which fixes the static attention problem of the standard GAT: since the linear layers in the standard GAT are applied right after each other, the ranking of attended nodes is unconditioned on the query node.  In contrast, in GATv2, every node can attend to any other node.

Thanks in advance.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,6 +82,7 @@ In detail, the following methods are currently implemented:
 * **[ECConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.ECConv)** from Simonovsky and Komodakis: [Edge-Conditioned Convolution on Graphs](https://arxiv.org/abs/1704.02901) (CVPR 2017)
 * **[EGConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.EGConv)** from Tailor *et al.*: [Adaptive Filters and Aggregator Fusion for Efficient Graph Convolutions](https://arxiv.org/abs/2104.01481) (GNNSys 2021) [[**Example**](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/egc.py)]
 * **[GATConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GATConv)** from Veličković *et al.*: [Graph Attention Networks](https://arxiv.org/abs/1710.10903) (ICLR 2018) [[**Example**](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/gat.py)]
+* **[GATv2Conv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GATv2Conv)** from Brody *et al.*: [How Attentive are Graph Attention Networks?](https://arxiv.org/abs/2105.14491) (CoRR 2021)
```
Comment (rusty1s) [inline README.md:85]: How about we name this one `GAT2Conv`, similar to `GCN2Conv`? Is that okay for you?

File: torch_geometric/nn/conv/gatv2_conv.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.utils import remove_self_loops, add_self_loops, softmax
+
+from torch_geometric.nn.inits import glorot, zeros
+
+
+class GATv2Conv(MessagePassing):
+    r"""The GATv2 operator from the `"How Attentive are Graph Attention Networks?"
+    <https://arxiv.org/abs/2105.14491>`_ paper, which fixes the static
+    attention problem of the standard GAT: since the linear layers in
```
Comment (rusty1s) [inline torch_geometric/nn/conv/gatv2_conv.py:18]: of the standard :class:\`~torch_geometric.conv.GATConv\` layer:

File: torch_geometric/nn/conv/gatv2_conv.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        dropout (float, optional): Dropout probability of the normalized
+            attention coefficients which exposes each node to a stochastically
+            sampled neighborhood during training. (default: :obj:`0`)
+        add_self_loops (bool, optional): If set to :obj:`False`, will not add
+            self-loops to the input graph. (default: :obj:`True`)
+        bias (bool, optional): If set to :obj:`False`, the layer will not learn
+            an additive bias. (default: :obj:`True`)
+        share_weights (bool, optional): If set to :obj:`True`, the same matrix
+            will be applied to the source and the target node of every edge.
+        (default: :obj:`False`)
```
Comment (rusty1s) [inline torch_geometric/nn/conv/gatv2_conv.py:59]: align with previous row

File: torch_geometric/nn/conv/gatv2_conv.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        \exp\left(\mathbf{a}^{\top}\mathrm{LeakyReLU}\left(\mathbf{\Theta}
+        [\mathbf{x}_i \, \Vert \, \mathbf{x}_j]
+        \right)\right)}
+        {\sum_{k \in \mathcal{N}(i) \cup \{ i \}}
+        \exp\left(\mathbf{a}^{\top}\mathrm{LeakyReLU}\left(\mathbf{\Theta}
+        [\mathbf{x}_i \, \Vert \, \mathbf{x}_k]
+        \right)\right)}.
+
+    Args:
+        in_channels (int): Size of each input sample.
```
Comment (rusty1s) [inline torch_geometric/nn/conv/gatv2_conv.py:41]: `int` or `tuple`

**Review by rusty1s — APPROVED** [review]
Thanks a lot. This is super cool! I only added some minor comments :)

File: README.md (hunk 1 above)
Comment (shakedbr) [reply README.md:85]: To be consistent with the paper's notation I rather stay with `GATv2Conv`

**Comment by shakedbr** [comment]
Thanks, @rusty1s!

Fixed the small issues

Please check if it's ready for merging.

**Comment by rusty1s** [comment]
Looks great. Thanks a lot for fixing :)

**Pull request merged**